test/Microsoft.ML.StaticPipelineTesting/StaticPipeTests.cs

```diff
@@ -327,5 +327,43 @@ public void NormalizerWithOnFit()
                 Console.WriteLine(Encoding.UTF8.GetString(stream.ToArray()));
             }
         }
+
+        [Fact]
+        public void ToKey()
```

Comment:
>ToKey [](start = 20, length = 5)

Do you keep insisting that all the pigsty testing is done in this file?

Reply:
No, not insisting, it is just useful to me for this to be the case since as far as I see visual studio lost the ability to group tests by anything other than the class, and I am often wanting to run all pigsty using tests (to ensure that when I change the infrastructure I don't break things). I would be more enthusiastic about putting the tests somewhere esle if there was some mechanism to select all tests of a given category, which regrettably I do not see is possible.

---
In reply to: [216414931](https://github.com/dotnet/machinelearning/pull/870#discussion_r216414931) [](ancestors = 216414931)

Reply:
This may be something we want to bring up -- we used to have tests categories pretty well lined up across the different things you might want to test -- but now that mechanism seems to be gone, at least from the VS UI...

---
In reply to: [216490385](https://github.com/dotnet/machinelearning/pull/870#discussion_r216490385) [](ancestors = 216490385,216414931)